c_glib/gandiva-glib/node.cpp

```diff
@@ -1139,49 +1169,56 @@ ggandiva_literal_node_new_raw(std::shared_ptr<gandiva::Node> *gandiva_node)
-    type = GGANDIVA_TYPE_BOOLEAN_LITERAL_NODE;
-    break;
-  case arrow::Type::type::UINT8:
-    type = GGANDIVA_TYPE_UINT8_LITERAL_NODE;
-    break;
-  case arrow::Type::type::UINT16:
-    type = GGANDIVA_TYPE_UINT16_LITERAL_NODE;
-    break;
-  case arrow::Type::type::UINT32:
-    type = GGANDIVA_TYPE_UINT32_LITERAL_NODE;
-    break;
-  case arrow::Type::type::UINT64:
-    type = GGANDIVA_TYPE_UINT64_LITERAL_NODE;
-    break;
-  case arrow::Type::type::INT8:
-    type = GGANDIVA_TYPE_INT8_LITERAL_NODE;
-    break;
-  case arrow::Type::type::INT16:
-    type = GGANDIVA_TYPE_INT16_LITERAL_NODE;
-    break;
-  case arrow::Type::type::INT32:
-    type = GGANDIVA_TYPE_INT32_LITERAL_NODE;
-    break;
-  case arrow::Type::type::INT64:
-    type = GGANDIVA_TYPE_INT64_LITERAL_NODE;
-    break;
-  case arrow::Type::type::FLOAT:
-    type = GGANDIVA_TYPE_FLOAT_LITERAL_NODE;
-    break;
-  case arrow::Type::type::DOUBLE:
-    type = GGANDIVA_TYPE_DOUBLE_LITERAL_NODE;
-    break;
-  case arrow::Type::type::STRING:
-    type = GGANDIVA_TYPE_STRING_LITERAL_NODE;
-    break;
-  case arrow::Type::type::BINARY:
-    type = GGANDIVA_TYPE_BINARY_LITERAL_NODE;
-    break;
-  default:
-    type = GGANDIVA_TYPE_LITERAL_NODE;
-    break;
+  auto gandiva_literal_node =
+    std::static_pointer_cast<gandiva::LiteralNode>(*gandiva_node);
+
+  if (gandiva_literal_node->is_null()) {
+    type = GGANDIVA_TYPE_NULL_LITERAL_NODE;
+  } else {
+    switch ((*gandiva_node)->return_type()->id()) {
+    case arrow::Type::BOOL:
+      type = GGANDIVA_TYPE_BOOLEAN_LITERAL_NODE;
+      break;
+    case arrow::Type::type::UINT8:
+      type = GGANDIVA_TYPE_UINT8_LITERAL_NODE;
+      break;
+    case arrow::Type::type::UINT16:
+      type = GGANDIVA_TYPE_UINT16_LITERAL_NODE;
+      break;
+    case arrow::Type::type::UINT32:
+      type = GGANDIVA_TYPE_UINT32_LITERAL_NODE;
+      break;
+    case arrow::Type::type::UINT64:
+      type = GGANDIVA_TYPE_UINT64_LITERAL_NODE;
+      break;
+    case arrow::Type::type::INT8:
+      type = GGANDIVA_TYPE_INT8_LITERAL_NODE;
+      break;
+    case arrow::Type::type::INT16:
+      type = GGANDIVA_TYPE_INT16_LITERAL_NODE;
+      break;
+    case arrow::Type::type::INT32:
+      type = GGANDIVA_TYPE_INT32_LITERAL_NODE;
+      break;
+    case arrow::Type::type::INT64:
+      type = GGANDIVA_TYPE_INT64_LITERAL_NODE;
+      break;
+    case arrow::Type::type::FLOAT:
+      type = GGANDIVA_TYPE_FLOAT_LITERAL_NODE;
+      break;
+    case arrow::Type::type::DOUBLE:
+      type = GGANDIVA_TYPE_DOUBLE_LITERAL_NODE;
+      break;
+    case arrow::Type::type::STRING:
+      type = GGANDIVA_TYPE_STRING_LITERAL_NODE;
+      break;
+    case arrow::Type::type::BINARY:
+      type = GGANDIVA_TYPE_BINARY_LITERAL_NODE;
+      break;
+    default:
+      type = GGANDIVA_TYPE_LITERAL_NODE;
+      break;
+    }
   }
   auto literal_node = GGANDIVA_LITERAL_NODE(g_object_new(type,
                                                          "node", gandiva_node,
```

Comment:
Can we add `"return-type", ...` here to reuse `return-type` property instead of defining `ggandiva_node_get_return_type()`?
I don't want to recreate `GArrowDataType` objects if it's available.

Reply:
I see. I'll reuse `return_type` property in `GGandivaLiteralNode` and `GGandivaFunctionNode`.